### ISSUE TITLE
After clicking an existing intervention, using the polygon draw tool doesn't work

### ISSUE THREAD

**Issue by Sparrow0hawk** [acteng/atip#107]
When using the current version of atip I find that if I click an existing intervention and then try and use the Polygon draw tool it doesn't work. I can get it working by clicking the Marker tool and clicking immediately back to the Polygon tool or clicking off a selected intervention and going back to the Polygon tool.

https://user-images.githubusercontent.com/20887250/229309797-ccde749e-7be8-448a-b4af-d02d7a885427.mov

I can actually reproduce this behaviour in prod [atip](https://atip.uk/) although the errors you see in the console actually relate to #106 

This clearly isn't a major issue as people have used prod atip where this behaviour occurs and not mentioned this to us (AFAIK), but potentially something we could do here to smooth out this particular experience.

**Comment by dabreegster** [comment]
I think I've hit this problem before in prod as well... I'm not sure what's going on. Maybe because we do `drawControls.changeMode` at different points to select an object in mapbox-gl-draw's perspective when clicking in the sidebar?

I'm in the middle of rewriting the world in https://github.com/acteng/atip/compare/main...draw_v3. This will still use mapbox-gl-draw, but in a very limited way -- just drop and drag a marker, and to draw and move vertices of a polygon. Switching modes (the toolbox on the right), rendering finalized interventions, and handling hover/clicking on finalized interventions will all be us. I think we need/want this level of control for other reasons, and hopefully it can let us fix bugs like this. (Or at least take the blame directly :) )

Depending how quickly I can get the new branch solidified, do you think it's worth prioritizing a fix in prod to the current behavior, if everybody has figured out the workaround intuitively?

**Comment by Sparrow0hawk** [comment]
I'd say not worth prioritizing a fix in prod at this time. My understanding is that because the ATF funding round is over, use of atip.uk will be minimal at the moment (@Robinlovelace can correct me!) so think we can afford to leave this as you're rewriting the world!

**Comment by robinlovelace-ate** [comment]
> My understanding is that because the ATF funding round is over, use of atip.uk will be minimal at the moment (@Robinlovelace can correct me!) so think we can afford to leave this as you're rewriting the world!

Agreed. We're in go fast break stuff mode. We're working behind the scenes to get future proof hosting solution building on comments in #104, the key policy use case has been delivered, and there is no immediate use planned. I plan to look at the logs out of interest: although we're not putting it out it's a very handy public tool so I wouldn't be surprise if usage grows without us putting it out there.

One thought: worth putting a "ATIP is undergoing development and may change without notice" type landing page, similar in spirit to the landing message on https://www.npt.scot?

**Comment by dabreegster** [comment]
> One thought: worth putting a "ATIP is undergoing development and may change without notice" type landing page

I think that'd be helpful. We could put it on the frontpage (area selection) and in the About button on the per-area page. (Let's please not interrupt the flow with a mandatory popup or anything!) It could also be helpful to have a very rough changelog shown here, like "April 5 - new draw controls, April 8 - new form... click here to give feedback"

**Comment by robinlovelace-ate** [comment]
Update on usage: very low since ATF4 submission, good to see everyone beavering away after my demo of the tool and in the rush to get their bids submitted (then relax)!

![image](https://user-images.githubusercontent.com/122299965/229345870-1c728ad6-c9c5-4ecf-bbce-523008608c97.png)

**Comment by robinlovelace-ate** [comment]
Basically: no users now so we should focus development effort on future proof stuff, that may include NEWs and notice in about to expect changes but the data visualised above emphasises that we can break stuff without too much worry. One other question: how easy would it be to serve each stable release, e.g. v0.0.1.atip.uk always has contents of release `v0.0.1`?

**Comment by dabreegster** [comment]
The bug described here is fixed in github main (but not prod), but there's good ideas here about adding a changelog to the landing page, so retitling the issue
